### PR TITLE
Insets library

### PR DESCRIPTION
**Pull request by chrisbanes**
This PR imports https://goo.gle/compose-insets into Accompanist, to make usage easier.

### CODE REVIEW

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Insets.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+@Composable
+fun ProvideWindowInsets(
+    consumeWindowInsets: Boolean = true,
+    content: @Composable () -> Unit
+) {
+    val view = ViewAmbient.current
+
+    val windowInsets = remember { WindowInsets() }
+
+    onCommit(view) {
```
Comment (nickbutcher) [inline insets/src/main/java/dev/chrisbanes/accompanist/insets/Insets.kt:123]: Move this to `DisposableEffect`?

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+/**
+ * Returns the current insets converted into a [PaddingValues].
+ *
+ * @param start Whether to apply the inset on the start dimension.
+ * @param top Whether to apply the inset on the top dimension.
+ * @param end Whether to apply the inset on the end dimension.
+ * @param bottom Whether to apply the inset on the bottom dimension.
+ */
+@Composable
+fun Insets.toPaddingValues(
```
Comment (nickbutcher) [inline insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt:162]: Does this belong in Size or Padding 🤔

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+            else -> 0.dp
+        },
+        bottom = when {
+            bottom -> this@toPaddingValues.bottom.toDp()
+            else -> 0.dp
+        }
+    )
+}
+
+private data class InsetsSizeModifier(
```
Comment (nickbutcher) [inline insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt:191]: Comment

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Insets.kt (hunk 1 above)
Comment (chrisbanes) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/Insets.kt:123]: Good call. Done.

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt (hunk 2 above)
Comment (chrisbanes) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt:162]: Yeah, missed this when copying. Moved.

File: insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt (hunk 3 above)
Comment (chrisbanes) [reply insets/src/main/java/dev/chrisbanes/accompanist/insets/Size.kt:155]: Done!

File: .idea/codeStyles/Project.xml [hunk 4]
```diff
@@ -2,7 +2,11 @@
   <code_scheme name="Project" version="173">
     <codeStyleSettings language="kotlin">
       <option name="PACKAGES_TO_USE_STAR_IMPORTS">
-        <value />
+        <value>
```
Comment (nickbutcher) [inline .idea/codeStyles/Project.xml:5]: Revert these changes?

File: insets/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+dependencies {
+    implementation "dev.chrisbanes.accompanist:accompanist-insets:<version>"
+}
+```
+
+Snapshots of the development version are available in [Sonatype's `snapshots` repository][snap]. These are updated on every commit.
+
+[compose]: https://developer.android.com/jetpack/compose
+[snap]: https://oss.sonatype.org/content/repositories/snapshots/dev/chrisbanes/accompanist/accompanist-glide/
```
Comment (nickbutcher) [inline insets/README.md:107]:
```suggestion
[snap]: https://oss.sonatype.org/content/repositories/snapshots/dev/chrisbanes/accompanist/accompanist-insets/
```

**Review by nickbutcher — APPROVED** [review]
